### PR TITLE
dropbox workitem

### PR DESCRIPTION
**Pull request by tsurdilo**

### CODE REVIEW

**Comment by tsurdilo** [comment]
@mswiderski here comes another one :)
-- dropbox file upload and file download workitem

**Comment by tsurdilo** [comment]
will need to be rebased little after the ibm watson one gets merged (or the other way around whichever goes first)

File: dropbox-workitem/src/main/java/org/jbpm/process/workitem/dropbox/DownloadFileWorkitemHandler.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+            doc.setIdentifier(dropboxPath + "/" + documentName);
+            doc.setLastModified(new Date());
+            doc.setContent(IOUtils.toByteArray(inStream));
+            doc.setSize(inStream.available());
+
+            results.put(RESULTS_DOCUMENT,
+                        doc);
+        } catch (Exception e) {
+            logger.error("Unable to download file: " + e.getMessage());
+            abortWorkItem(workItem,
```
Comment (mswiderski) [inline dropbox-workitem/src/main/java/org/jbpm/process/workitem/dropbox/DownloadFileWorkitemHandler.java]: abortWorkItem should not be called here, more we should handleException instead

File: dropbox-workitem/src/main/java/org/jbpm/process/workitem/dropbox/DownloadFileWorkitemHandler.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+                          workItemManager);
+        }
+
+        workItemManager.completeWorkItem(workItem.getId(),
```
Comment (mswiderski) [inline dropbox-workitem/src/main/java/org/jbpm/process/workitem/dropbox/DownloadFileWorkitemHandler.java]: I would move this within try instead

File: dropbox-workitem/src/main/java/org/jbpm/process/workitem/dropbox/DownloadFileWorkitemHandler.java [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+            InputStream inStream = client.files().downloadBuilder(dropboxPath + "/" + documentName)
+                    .start().getInputStream();
+
+            Document doc = new DocumentImpl();
+            doc.setName(documentName);
+            doc.setIdentifier(dropboxPath + "/" + documentName);
+            doc.setLastModified(new Date());
+            doc.setContent(IOUtils.toByteArray(inStream));
+            doc.setSize(inStream.available());
```
Comment (mswiderski) [inline dropbox-workitem/src/main/java/org/jbpm/process/workitem/dropbox/DownloadFileWorkitemHandler.java]: is this still valid once the stream was already read?

File: dropbox-workitem/src/main/java/org/jbpm/process/workitem/dropbox/DownloadFileWorkitemHandler.java [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
+            if (auth == null) {
+                auth = new DropboxAuth();
+            }
+
+            client = auth.authorize(clientIdentifier,
+                                    accessToken);
+
+            String dropboxPath = (String) workItem.getParameter("Path");
+            String documentName = (String) workItem.getParameter("DocumentName");
```
Comment (mswiderski) [inline dropbox-workitem/src/main/java/org/jbpm/process/workitem/dropbox/DownloadFileWorkitemHandler.java]: wouldn't it be better to have single param "path" that points at file to download?

File: dropbox-workitem/src/main/java/org/jbpm/process/workitem/dropbox/UploadFileWorkitemHandler.java [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
+            ByteArrayInputStream docStream = new ByteArrayInputStream(documentToUpload.getContent());
+
+            client.files().uploadBuilder(dropboxPath + "/" + documentToUpload.getName())
+                    .withMode(WriteMode.ADD)
+                    .withClientModified(documentToUpload.getLastModified())
+                    .uploadAndFinish(docStream);
+        } catch (Exception e) {
+            logger.error("Unable to upload file: " + e.getMessage());
+            abortWorkItem(workItem,
```
Comment (mswiderski) [inline dropbox-workitem/src/main/java/org/jbpm/process/workitem/dropbox/UploadFileWorkitemHandler.java]: same here, no need to call abortWorkItem

File: dropbox-workitem/src/main/java/org/jbpm/process/workitem/dropbox/UploadFileWorkitemHandler.java [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+                          workItemManager);
+        }
+
+        workItemManager.completeWorkItem(workItem.getId(),
```
Comment (mswiderski) [inline dropbox-workitem/src/main/java/org/jbpm/process/workitem/dropbox/UploadFileWorkitemHandler.java]: move this within try block

**Comment by tsurdilo** [comment]
@mswiderski updated - thank you!

**Comment by tsurdilo** [comment]
rebased

**Pull request merged**